lndr-backend/src/Lndr/EthereumInterface.hs

```diff
@@ -133,40 +144,33 @@ interpretUcacLog change = do
+    transactionM <- lndrWeb3 . Eth.getTransactionByHash $ addHexPrefix txHash
+    case transactionM of
+        (Just transaction) ->
             let fromMatch = txFrom transaction == creditor creditRecord
                 toMatch = txTo transaction == Just (debtor creditRecord)
                 transactionValue = hexToInteger $ txValue transaction
                 settlementValue = fromMaybe 0 $ settlementAmount creditRecord
                 valueMatch = transactionValue == settlementValue
                 creditHash = T.unpack $ hash creditRecord
             in case (fromMatch, toMatch, valueMatch) of
-                (False, _, _)      -> pure . Left $ "Bad from match, hash: " ++ creditHash
-                (_, False, _)      -> pure . Left $ "Bad to match, hash: " ++ creditHash
-                (_, _, False)      -> pure . Left $ "Bad value match, hash: " ++ creditHash
+                (False, _, _)      -> lndrError $ "Bad from match, hash: " ++ creditHash
+                (_, False, _)      -> lndrError $ "Bad to match, hash: " ++ creditHash
+                (_, _, False)      -> lndrError $ "Bad value match, hash: " ++ creditHash
                                                  ++ "tx value: " ++ show transactionValue
                                                  ++ ", settlementValue: " ++ show settlementValue
-                (True, True, True) -> pure $ Right ()
-        _ -> pure . Left $ "transaction not found, tx_hash: " ++ T.unpack txHash
-verifySettlementPayment _ = pure $ Left "Incompelete settlement record"
-
-
-calculateSettlementCreditRecord :: ServerConfig -> CreditRecord -> CreditRecord
-calculateSettlementCreditRecord _ cr@(CreditRecord _ _ _ _ _ _ _ _ _ _ Nothing _) = cr
-calculateSettlementCreditRecord config cr@(CreditRecord _ _ amount _ _ _ _ _ ucac _ (Just currency) _) =
-    let blockNumber = latestBlockNumber config
-        prices = ethereumPrices config
-        currencyPerEth = case B.lookupR ucac (lndrUcacAddrs config) of
-            Just "USD" -> usd prices * 100 -- mutliplying by 100 here since
-                                           -- usd amounts are stored in cents
-            Just "JPY" -> jpy prices
-            Just "KRW" -> krw prices
-            Nothing    -> error "ucac not found"
-        settlementAmountRaw = floor $ fromIntegral amount / currencyPerEth * 10 ^ 18
-    in cr { settlementAmount = Just $ roundToMegaWei settlementAmountRaw
-          , settlementBlocknumber = Just blockNumber
-          }
+                (True, True, True) -> pure ()
+        Nothing -> lndrError $ "transaction not found, tx_hash: " ++ T.unpack txHash
+verifySettlementPayment _ = lndrError "Incompelete settlement record"
+
+
+-- | Queries the blockchain for current blocknumber.
+currentBlockNumber :: ServerConfig -> MaybeT IO Integer
+currentBlockNumber config = do
+    let provider = HttpProvider (web3Url config)
+    blockNumberTextE <- runWeb3' provider Eth.blockNumber
```

Comment:
Can the newly introduced `lndrWeb3` be used here instead? The logic in here is pretty similar, except the way the configuration is injected is different.

Reply:
Yeah, I thought the same. Will refactor this if possible, but it'll probably have to stay like this.